From b25807c839b05dd9ddcb09ed71a77138ebb82cf8 Mon Sep 17 00:00:00 2001
From: Vignesh Viswanathan <vignesh.viswanathan@oss.qualcomm.com>
Date: Thu, 4 Jun 2026 17:14:34 +0530
Subject: [PATCH] drivers: mhi: host: Increase RDDM timeout to 1s

Current rddm_timeout of 400ms is not sufficient for QCN9625 as the
device takes more time to complete the required crashdump callback
functions and switch to RDDM EE.

Increase the RDDM timeout to 1s to ensure device switches to RDDM.

Change-Id: I2742f631be7a76bdd497203319409f54bf6700e0
Signed-off-by: Vignesh Viswanathan <vignesh.viswanathan@oss.qualcomm.com>
---
 drivers/bus/mhi/host/boot.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/bus/mhi/host/boot.c b/drivers/bus/mhi/host/boot.c
index f01c397900b7..55a9989740fa 100644
--- a/drivers/bus/mhi/host/boot.c
+++ b/drivers/bus/mhi/host/boot.c
@@ -103,7 +103,7 @@ static int __mhi_download_rddm_in_panic(struct mhi_controller *mhi_cntrl)
 	const u32 delayus = 2000;
 	const u32 soc_reset_delay_ms = 200;
 	u32 retry = (mhi_cntrl->timeout_ms * 1000) / delayus;
-	const u32 rddm_timeout_us = 400000;
+	const u32 rddm_timeout_us = 1000000;
 	int rddm_retry = rddm_timeout_us / delayus;
 	void __iomem *base = mhi_cntrl->bhie;
 	struct device *dev = &mhi_cntrl->mhi_dev->dev;
-- 
2.34.1

